src/Microsoft.AspNet.Mvc.Razor/RazorPageActivator.cs

```diff
@@ -111,20 +112,29 @@ private PropertyActivator<ViewContext> CreateActivateInfo(PropertyInfo property)
             {
                 valueAccessor = context => context.ViewData;
             }
+            else if (typeof(IUrlHelper).IsAssignableFrom(property.PropertyType))
+            {
+                valueAccessor = context =>
+                {
+                    var serviceProvider = context.HttpContext.RequestServices;
+                    var factory = (IUrlHelperFactory)serviceProvider.GetRequiredService(typeof(IUrlHelperFactory));
+                    return factory.Create(context);
+                };
```

Comment:
This is already part of RazorPage by default. This is necessary to keep it working.